ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala

```diff
@@ -180,6 +180,15 @@ final class CommandsValidator(ledgerId: LedgerId, submissionIdGenerator: Submiss
       case ProtoEmpty =>
         Left(missingField("command"))
     }
+
+  private def matchSubmissionId(commands: ProtoCommands) = {
```

Comment:
I'd recommend `determineSubmissionId` or `takeOrGenerateSubmissionId` or similar name for this helper function ('match' implies we're using some kind of a mapping but we aren't).

Reply:
Perhaps `extractOrGenerateSubmissionId`?

Reply:
Fully aggree, will update to `extractOrGenerateSubmissionId`